src/main/java/uk/gov/companieshouse/acsp/manage/users/controller/UserAcspMembershipInternal.java

```diff
@@ -3,43 +3,89 @@
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.util.List;
+import java.util.Optional;
 
 @RestController
 public class UserAcspMembershipInternal implements UserAcspMembershipInternalInterface {
 
-    public UserAcspMembershipInternal() {}
-
-    @Override
-    public ResponseEntity<ResponseBodyPost> addAcspOwner(
-            @NotNull String xRequestId,
-            @Pattern(regexp = "^[0-9A-Za-z-_]{0,32}$") String acspNumber,
-            @Valid InternalRequestBodyPost internalRequestBodyPost) {
-        return null; // TODO(https://companieshouse.atlassian.net/browse/IDVA6-1238)
-    }
-
-    @Override
-    public ResponseEntity<Boolean> isActiveMember(
-            @NotNull String xRequestId,
-            @Pattern(regexp = "^[0-9A-Za-z-_]{0,32}$") String acspNumber,
-            @NotNull @Pattern(regexp = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$") String userEmail
-    ) {
-        return null; // TODO(https://companieshouse.atlassian.net/browse/IDVA6-1212)
-    }
-
-    @Override
-    public ResponseEntity<Void> performActionOnAcsp(
-            @NotNull String xRequestId,
-            @Pattern(regexp = "^[0-9A-Za-z-_]{0,32}$") String acspNumber,
-            @Valid InternalRequestBodyPatch internalRequestBodyPatch
-    ) {
-        return null;
-    }
+  private static final Logger LOG =
+      LoggerFactory.getLogger(StaticPropertyUtil.APPLICATION_NAMESPACE);
+  private final AcspDataService acspDataService;
+  private final UsersService usersService;
+  private final AcspMembersService acspMembersService;
 
-}
+  @Autowired
+  public UserAcspMembershipInternal(
+      AcspDataService acspDataService,
+      UsersService usersService,
+      AcspMembersService acspMembersService) {
+    this.acspDataService = acspDataService;
+    this.usersService = usersService;
+    this.acspMembersService = acspMembersService;
+  }
+
+  @Override
+  public ResponseEntity<ResponseBodyPost> addAcspOwner(
+      final String xRequestId,
+      final String acspNumber,
+      final InternalRequestBodyPost internalRequestBodyPost) {
+    LOG.debugContext(
+        xRequestId,
+        String.format("Attempting to add ACSP owner for acsp_number %s", acspNumber),
+        null);
+    acspDataService.fetchAcspData(acspNumber);
+    String userEmail = internalRequestBodyPost.getOwnerEmail();
+    UsersList users =
+        Optional.ofNullable(usersService.searchUserDetails(List.of(userEmail)))
+            .filter((userList -> !userList.isEmpty()))
+            .orElseThrow(
+                () -> {
+                  LOG.error(String.format("Failed to find user with userEmail %s", userEmail));
+                  return new NotFoundRuntimeException(
+                      "acsp-manage-users-api", "Failed to find user");
+                });
+    String userId = users.getFirst().getUserId();
+    final var acspMembers = acspMembersService.createAcspMembersWithOwnerRole(acspNumber, userId);
```

Comment:
before creating a new acsp we should check if there is an existing acsp with the same user , acsp number.
if the acsp exists, and its still active reject the request. 400, acsp already exists.
if the acsp exists, and its removed then create  it.